src/mmda/recipes/recipe.py

```diff
@@ -0,0 +1,62 @@
+"""
+
+
+@kylel
+
+"""
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+from mmda.types import *
+from mmda.parsers.pdfplumber_parser import PDFPlumberParser
+from mmda.rasterizers.rasterizer import PDF2ImageRasterizer
+from mmda.predictors.heuristic_predictors.dictionary_word_predictor import DictionaryWordPredictor
+from mmda.predictors.lp_predictors import LayoutParserPredictor
+from mmda.predictors.hf_predictors.token_classification_predictor import \
+    IVILATokenClassificationPredictor
+
+
+class Recipe:
+    def run(self, doc: Document) -> Document:
```

Comment:
`run` here takes a document, but `run` in `CoreRecipe` takes a path. If we are expecting different recipes to operate on different inputs, I'd split `run` out in different methods.

Reply:
fixed